developer-docs-site/docs/move/move-on-aptos/cryptography.md

```diff
@@ -0,0 +1,211 @@
+---
+title: "Cryptography"
+slug: "cryptography"
+---
+
+# Cryptography in Move
+
+The Aptos adapter for Move has introduced many useful cryptographic primitives:
+
+1. Cryptographic hash functions (e.g., SHA2-256, SHA3-256, Keccak256, Blake2b-256)
+2. Digital signature verification (e.g., Ed25519, ECDSA, BLS)
+3. Elliptic curve arithmetic (e.g., Ristretto255, BLS12-381)
+4. Groth16 zero-knowledge proof (ZKP) verification
+5. Bulletproofs zero-knowledge range proof (ZKRP) verification
+
+The **key guiding principles** behind these cryptographic extensions are:
+
+1. **Minimize gas costs** for Move developers by implementing key primitives as [Move native functions](./move/book/functions#native-functions) (e.g., see the [BLS signatures over BLS12-381 elliptic curves](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-stdlib/sources/cryptography/bls12381.move) module)
+2. Type-safe APIs (e.g., see the [Ed25519 signature module](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-stdlib/sources/cryptography/ed25519.move))
+3. When native functions are not available, **empower developers** to implement their own cryptographic primitives on top of abstract cryptographic building blocks such as _finite fields_ and _Abelian groups_ (e.g., via the [`aptos_std::crypto_algebra`](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-stdlib/sources/cryptography/crypto_algebra.move) module).
+
+An overview of these features and (some of) the applications they enable follows below. 
+For a detailed view, see the [cryptography Move modules code](https://github.com/aptos-labs/aptos-core/tree/main/aptos-move/framework/aptos-stdlib/sources/cryptography).
+
+## Cryptographic hash functions in Move
+
+Developers can now use more cryptographic hash functions in Move via the [`aptos_std::aptos_hash`](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-stdlib/sources/hash.move) module:
+
+| Hash function | Hash size (bits) | Collision-resistance security (bits) |
+|---------------|------------------|--------------------------------------|
+| Keccak256     | 256              | 128                                  |
+| SHA2-512      | 512              | 256                                  |
+| SHA3-512      | 512              | 256                                  |
+| RIPEMD160     | 160              | 80 (**weak**)                        |
+| Blake2b-256   | 256              | 128                                  |
```

Comment:
could we add in some rough gas usage per ... 1KB or something?
and maybe even potential applications?

Reply:
We will do this later, once @mstraka100 figures out an easy way to benchmark gas costs.